pkg/agent/openflow/network_policy.go

```diff
@@ -1040,15 +1030,35 @@ func (c *client) UninstallPolicyRuleFlows(ruleID uint32) ([]string, error) {
 		return nil, err
 	}
 
-	for _, p := range ofPrioritiesToUninstallFlows {
-		conjsStalePriority, _ := c.policyCache.ByIndex(priorityIndex, p)
-		if len(conjsStalePriority) == 0 {
+	c.policyCache.Delete(conj)
+	return staleOFPriorities, nil
+}
+
+// getStalePriorities returns the ofPriorities that will be stale on the rule table where the
+// policyRuleConjunction is installed, after the deletion of that policyRuleConjunction.
+func (c *client) getStalePriorities(conj *policyRuleConjunction) (staleOFPriorities []string) {
+	var ofPrioritiesPotentiallyStale []string
+	if conj.ruleTableID != IngressRuleTable && conj.ruleTableID != EgressRuleTable {
+		ofPrioritiesPotentiallyStale = conj.ActionFlowPriorities()
+	}
+	klog.V(4).Infof("Potential stale ofpriority %v found", ofPrioritiesPotentiallyStale)
+	for _, p := range ofPrioritiesPotentiallyStale {
```

Comment:
Could you give complexity for this loop? Concerned on nested loops and operations like lookup by index and removing slice element.

Reply:
This is a 'fake' nested loop. As of now, all policyRuleConjuections (each correspond to a single NetworkPolicy rule) will only have one actionFlow, so conj.ActionFlowPriorities() will contain only one item.
 
The purpose of the function is that, whenever a CNP conjuction is deleted, we check if there are still any other conjunctions created _on the same table_ with the same ofPriority (which means those rules are of the same Tier/policy/rule priority). If not, we can release the ofPriority for that particular table.

What the indexer filters out is the conjunctions which are created the the same ofPriority, only for CNP conjunctions (who will be, in 99.9% cases, created at a different ofPriority compared to K8s actionFlows). After getting the list of conjuctions, we then cross-validate the table number. This is done because it seems indexer does not support indexing on two keys aggregately. But the computation impact of this loop should not be concerning.

Reply:
Refactored the function to get rid of list operations